Ruby/Ruby.sublime-syntax

```diff
@@ -106,16 +106,18 @@ contexts:
   comments:
     # multiline comments
     - match: ^=begin
-      scope: punctuation.definition.comment.ruby
+      scope: punctuation.definition.comment.begin.ruby
       push:
         - meta_scope: comment.block.documentation.ruby
         - match: ^=end
-          scope: punctuation.definition.comment.ruby
+          scope: punctuation.definition.comment.end.ruby
+          pop: true
+    - match: \#+
```

Comment:
I'm undecided on whether we should match this syntactically or semantically. Syntactically, only the first `#` is the actual punctuation, but semantically multiple glyphs can be used. This is the first time I'm seeing this.

Otoh, the `*` asterisk in continued c-style block comments recently received a punctuation scope as well.

Reply:
Just considder the following comment

```
#########
# the text
#########
```

I wouldn't like to only see the first `#` but all in their own color. Anything else would look odd, I think.

Reply:
I understand that. I'm just saying this is unprecedented.

Reply:
These lines are basically just a copy of Perl's comments.

https://github.com/sublimehq/Packages/blob/d1494f4a5a2bf8c6ae2f3e7ad762a05273df97ef/Perl/Perl.sublime-syntax#L88-L93

Reply:
For the record, I think applying the scope to all of the leading punctuation is probably what the user will generally want, so let's try it.